### ISSUE TITLE
Submit upstream

### ISSUE THREAD

**Issue by gsauthof** [trailofbits/tsc_freq_khz#2]
Having a `/sys/devices/system/cpu/cpu0/tsc_freq_khz` seems to be generally useful.

Do you plan to submit a patch with this functionality to the LKML?

**Comment by artemdinaburg** [comment]
I agree it would be useful but I think the current code is not ready: it is missing some important parts like checking for existence of CPU0 (not guaranteed with hotplug), and verifying that the value makes sense before exporting it (in the case of a tsc that doesn't sync across cores, for instance). 

I was hoping that Google could upstream whatever they have, it is probably much better than my version :).

**Comment by shuckc** [comment]
This is nice - much better than userspace calibration code. Thanks
